pkg/blockbuilder/scheduler/scheduler.go

```diff
@@ -2,43 +2,134 @@ package scheduler
 
 import (
 	"context"
+	"errors"
+	"flag"
+	"strconv"
+	"time"
 
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/services"
 	"github.com/grafana/loki/v3/pkg/blockbuilder/types"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/twmb/franz-go/pkg/kadm"
 )
 
 var (
 	_ types.Scheduler = unimplementedScheduler{}
-	_ types.Scheduler = &QueueScheduler{}
+	_ types.Scheduler = &BlockScheduler{}
 )
 
-// unimplementedScheduler provides default implementations that panic.
-type unimplementedScheduler struct{}
+type Config struct {
+	ConsumerGroup string
+	Interval      time.Duration
 
-func (s unimplementedScheduler) HandleGetJob(_ context.Context, _ string) (*types.Job, bool, error) {
-	panic("unimplemented")
+	TargetRecordConsumptionPeriod time.Duration
 }
 
-func (s unimplementedScheduler) HandleCompleteJob(_ context.Context, _ string, _ *types.Job) error {
-	panic("unimplemented")
+func (cfg *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	f.DurationVar(&cfg.Interval, prefix+"interval", time.Minute, "How often to run scheduler top compute jobs.")
+	f.DurationVar(&cfg.TargetRecordConsumptionPeriod, prefix+"target-records-spanning-period", time.Hour, "Period used by the planner to calculate the start and end offset such that each job consumes records spanning the target period.")
+	f.StringVar(&cfg.ConsumerGroup, prefix+"consumer-group", "block-scheduler", "Consumer group used by block scheduler to track the last consumed offset.")
 }
 
-func (s unimplementedScheduler) HandleSyncJob(_ context.Context, _ string, _ *types.Job) error {
-	panic("unimplemented")
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	cfg.RegisterFlagsWithPrefix("block-scheduler.", f)
+}
+
+func (cfg *Config) Validate() error {
+	if cfg.Interval.Seconds() <= 0 {
```

Comment:
nit: this can be misleading. If you want, you can use `> 0` b/c `time.Duration` is an `int64` internally. Otherwise, update the error msg to say "must be more than $min_threshold".